packages/graphql-function-transformer/src/FunctionTransformer.ts

```diff
@@ -0,0 +1,184 @@
+import { obj, str, ref, printBlock, compoundExpression, qref } from 'graphql-mapping-template';
+import { ResolverResourceIDs, FunctionResourceIDs, ResourceConstants } from 'graphql-transformer-common';
+import { ObjectTypeDefinitionNode, FieldDefinitionNode, DirectiveNode } from 'graphql';
+import { AppSync, IAM, Fn } from 'cloudform-types'
+import { lambdaArnResource } from './lambdaArns';
+
+const FUNCTION_DIRECTIVE_STACK = 'FunctionDirectiveStack';
+
+export default class FunctionTransformer extends Transformer {
+
+    constructor() {
+        super(
+            'FunctionTransformer', 
+            gql`directive @function(name: String!, region: String) on FIELD_DEFINITION`
+        )
+    }
+
+    /**
+     * Add the required resources to invoke a lambda function for this field.
+     */
+    field = (parent: ObjectTypeDefinitionNode, definition: FieldDefinitionNode, directive: DirectiveNode, ctx: TransformerContext) => {
+        const { name, region } = getDirectiveArguments(directive);
+        if (!name) {
+            throw new TransformerContractError(`Must supply a 'name' to @function.`)
+        }
+
+        // Add the iam role if it does not exist.
+        const iamRoleKey = FunctionResourceIDs.FunctionIAMRoleID(name, region);
+        if (!ctx.getResource(iamRoleKey)) {
+            ctx.setResource(iamRoleKey, this.role(name, region));
+            ctx.addToStackMapping(FUNCTION_DIRECTIVE_STACK, iamRoleKey);
+        }
+
+        // Add the data source if it does not exist.
+        const lambdaDataSourceKey = FunctionResourceIDs.FunctionDataSourceID(name, region);
+        if (!ctx.getResource(lambdaDataSourceKey)) {
+            ctx.setResource(lambdaDataSourceKey, this.datasource(name, region));
+            ctx.addToStackMapping(FUNCTION_DIRECTIVE_STACK, lambdaDataSourceKey);
+        }
+
+        // Add function that invokes the lambda function
+        const functionConfigurationKey = FunctionResourceIDs.FunctionAppSyncFunctionConfigurationID(name, region);
+        if (!ctx.getResource(functionConfigurationKey)) {
+            ctx.setResource(functionConfigurationKey, this.function(name, region));
+            ctx.addToStackMapping(FUNCTION_DIRECTIVE_STACK, functionConfigurationKey);
+        }
+
+        // Add resolver that invokes our function
+        const typeName = parent.name.value;
+        const fieldName = definition.name.value;
+        const resolverKey = ResolverResourceIDs.ResolverResourceID(typeName, fieldName);
+        const resolver = ctx.getResource(resolverKey);
+        if (!resolver) {
+            ctx.setResource(resolverKey, this.resolver(typeName, fieldName, name, region));
+            ctx.addToStackMapping(FUNCTION_DIRECTIVE_STACK, resolverKey);
+        } else if (resolver.Properties.Kind === 'PIPELINE') {
+            ctx.setResource(resolverKey, this.appendFunctionToResolver(resolver, FunctionResourceIDs.FunctionAppSyncFunctionConfigurationID(name, region)))
```

Comment:
This is the first transformer to implement pipeline resolvers. You may use @function twice on a field and a pipeline resolver will be created with two functions. The response mapping template will return the result of the last function in the pipeline. Previous results will be passed to each subsequent function via `event.prev` in the lambda function.